scripts/gis-material-update/process/modules/hsl.py

```diff
@@ -1,14 +1,381 @@
+        return feed
+
+    def feed(self) -> gk.Feed:
+        return self._feed
+
+    def transit_week(self) -> list[str]:
+        """Return date strings of transit week"""
+        return self._transit_week
+
+    def transit_day(self, day_sel: int = 1) -> str:
+        """Return date of transit week.
+
+        Default is date for tuesday.
+        """
+        if day_sel > 6:
+            day_sel = day_sel % 7
+
+        return self.transit_week()[day_sel]
+
+    def _pick_service_ids(self, d_start: str, d_end: str) -> pd.Index:
+        """Pick service_id:s that fall within provided range.
+
+        Return index to calendar table."""
+        result = self.feed().calendar
+        ind = result.apply(
+            lambda r: self._overlaps(d_start, d_end, r.start_date, r.end_date), axis=1
+        )
+        return ind
+
+    def _overlaps(self, d1min: str, d1max: str, d2min: str, d2max: str) -> bool:
+        """Check if date ranges overlap"""
+        return min(as_date(d1max), as_date(d2max)) - max(
+            as_date(d1min), as_date(d2min)
+        ) >= timedelta(0)
+
+    def _pick_service_ids_for_one_day(self, datestring: str) -> list[str]:
+        """Return list of service ids of a given day.
+        Does not respect calendar_dates -information."""
+        result_ind = self._pick_service_ids(datestring, datestring)
+        candidate = self.feed().calendar.copy()
+        weekday = as_date(datestring).strftime("%A").lower()
+        weekday_ind = candidate[weekday] == 1
+        # candidate day must belong to time range AND have specific weekday mentioned in calendar
+        serv_id_cand = candidate[result_ind & weekday_ind]
+        return serv_id_cand["service_id"].values.tolist()
+
+    def _trips_for_service_ids(self, service_ids: list[str]) -> pd.DataFrame:
+        """Pick all trip_id:s belonging to a list of provided service_ids"""
+        srv_ind = self.feed().trips["service_id"].isin(service_ids).values
+        return self.feed().trips[srv_ind]
+
+    def _rush_hours(self) -> list[tuple[str, time, time]]:
+        """Construct rush hour times.
+
+        Return list of (<descriptor string>, start time, end time)"""
+
+        hours = []
+        # rush hours are morning rush hours (13 ranges):
+        # 6.00 - 7.00
+        # 6.15 - 7.15
+        # ...
+        # 8.45 - 9.45
+        # 9.00 - 10.00
+        #
+        # and evening rush hours (13 ranges):
+        # 15.00 - 16.00
+        # 15.15 - 16.15
+        # ...
+        # 17.45 - 18.45
+        # 18.00 - 19.00
+        for s in [6, 15]:
+            start_times = []
+            for h in range(4):
+                for m in [0, 15, 30, 45]:
+                    # construct list of tuples (hour_start, minute_start, hour_end, minute_end)
+                    start_times.append((s + h, m, s + h + 1, m))
+            # pick first 13 values, according to spec commented above
+            start_times = start_times[0:13]
```

Comment:
It's not easy for the reader to see why taking the 13 first values is necesssary and why it produces the right set of times. 

It might be easier to add the last interval separately, if using `for h in range(3):` earlier.
```
            # Add 9.00 - 10.00 or 18.00 - 19.00
            start_times.append((s + 3, 0, s + 4, 0))
```
This would also mean that the start_times variable is not needed, since the values can be appended directly to hours.